docs/developer_guide/explanations/developer_conventions.rst

```diff
@@ -27,3 +27,166 @@ When updating dependencies or modifying `conda/meta.yaml`, it is important to te
+
+When creating an issue, please provide as much information as possible, including:
+
+* Steps to reproduce the issue if it is a bug.
+* A clear description of the feature you would like to see if it is a feature request. - remember to include the use case and argue to importance of the feature.
+* Any relevant screenshots or logs.
+* Any other information that you think may be useful.
+
+When creating an issue, please ensure that you assign the issue to yourself if you are going to work on it or leave the assignment blank if not. This will prevent multiple people from working on the same issue at the same time.
+
+Please also ensure that any relevant labels are added to the issue. This will help the Mantid Imaging team to prioritise the issue and ensure that it is worked on in a timely manner.
+
+Once you have completed the above, you will also need to point the ticket with a number which will act as a rough estimate for the time it will take to complete the issue in whole days (round up to the nearest day), including any necessary research someone may need to do and the time to review and test.
+
+Finally, you will need to assign the issue to the relevant project board and sprint if applicable in the "ToDo" column if you believe the issue is ready to be worked on in the `current sprint <https://github.com/orgs/mantidproject/projects/13/views/6>`_.
+
+If you are unsure about any of the above, please ask a member of the Mantid Imaging team for help.
+
+Branches
+~~~~~~~~
+
+Mantid Imaging treats the ``main`` branch as the default branch. This branch should always be in a working state and should be used as the base for all new features and bug fixes.
+
+When working on a new feature or bug fix, you should create a new development branch from the ``main`` branch after you have first checked that your local ``main`` branch is up to date with the remote ``main`` branch. This can be checked by running to following command:
+
+.. code::
+
+    git fetch origin main && git pull origin main
+
+
+* ``git fetch origin main`` will fetch the latest changes from the remote ``main`` branch of the origin repository, updating your local copy of the remote-tracking branch ``origin/main`` with the latest commits from the remote repository.
+
+* ``git pull origin main`` will then pull the latest changes from the remote ``main`` branch of the origin repository into your local ``main`` branch. 
+
+In summary, this command updates your local branch with the latest changes from the remote ``main`` branch. **You do not need to checkout the main branch to do this**.
+
+The branch should be named in the following format:
+
+.. code::
+
+    <issue_number>_short_description    # e.g. 1234_add_new_feature
+
+
+You can create a new branch by running the following command (replacing ``<issue_number>_short_description`` with the relevant information):
+
+.. code::
+
+    git checkout -b <issue_number>_short_description
```

Comment:
I prefer the `git switch` commands these days, e.g.

`git switch -c 1234_add_new_feature`

Reply:
Fair enough, I still prefer git checkout, but I don't have strong feelings towards using git switch and restore